spec/decorators/prosecution_case_decorator_spec.rb

```diff
@@ -43,50 +49,132 @@
     it { is_expected.to all(be_instance_of(HearingDecorator)) }
 
-    it {
-      expect(call.map(&:hearing_days))
-        .to match([hearing3.hearing_days, hearing1.hearing_days, hearing2.hearing_days])
-    }
-  end
+    context 'when sort_colum is date and direction is asc' do
+      before do
+        allow(decorator).to receive(:column).and_return('date')
+        allow(decorator).to receive(:direction).and_return('asc')
+      end
 
-  describe '#hearings_with_day_by_datetime' do
-    subject(:call) { decorator.hearings_with_day_by_datetime }
+      let(:expected_days) do
+        ['2021-01-18T11:00:00.000Z'.to_datetime,
+         '2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime,
+         '2021-01-20T16:00:00.000Z'.to_datetime]
+      end
 
-    before { allow(prosecution_case).to receive_messages(hearings: hearings) }
+      it { expect(call.map(&:day)).to eql(expected_days) }
+    end
 
-    let(:hearings) { [hearing1, hearing2, hearing3] }
-    let(:hearing1) { CourtDataAdaptor::Resource::Hearing.new(hearing_days: hearing1_days) }
-    let(:hearing2) { CourtDataAdaptor::Resource::Hearing.new(hearing_days: hearing2_days) }
-    let(:hearing3) { CourtDataAdaptor::Resource::Hearing.new(hearing_days: hearing3_days) }
-    let(:hearing1_days) { ['2021-01-19T10:45:00.000Z', '2021-01-20T10:45:00.000Z'] }
-    let(:hearing2_days) { ['2021-01-20T16:00:00.000Z'] }
-    let(:hearing3_days) { ['2021-01-18T11:00:00.000Z'] }
+    context 'when sort_colum is date and direction is desc' do
+      before do
+        allow(decorator).to receive(:column).and_return('date')
+        allow(decorator).to receive(:direction).and_return('desc')
+      end
+
+      let(:expected_days) do
+        ['2021-01-20T16:00:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime,
+         '2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-18T11:00:00.000Z'.to_datetime]
+      end
 
-    let(:expected_days) do
-      ['2021-01-18T11:00:00.000Z'.to_datetime,
-       '2021-01-19T10:45:00.000Z'.to_datetime,
-       '2021-01-20T10:45:00.000Z'.to_datetime,
-       '2021-01-20T16:00:00.000Z'.to_datetime]
+      it { expect(call.map(&:day)).to eql(expected_days) }
     end
 
-    it { is_expected.to be_instance_of(Enumerator) }
-    it { is_expected.to all(be_instance_of(HearingDecorator)) }
-    it { expect(call.map(&:day)).to contain_exactly(*expected_days) }
+    context 'when sort_colum is type and direction is asc' do
+      before do
+        allow(decorator).to receive(:column).and_return('type')
+        allow(decorator).to receive(:direction).and_return('asc')
+      end
+
+      let(:expected_days) do
+        ['2021-01-20T16:00:00.000Z'.to_datetime,
+         '2021-01-18T11:00:00.000Z'.to_datetime,
+         '2021-01-19T10:45:00.000Z'.to_datetime,
+         '2021-01-20T10:45:00.000Z'.to_datetime]
+      end
+
+      it { expect(call.map(&:day)).to eql(expected_days) }
+    end
```

Comment:
this "says" it is testing `type` sorting but is testing based on date/day. would be confusing

Reply:
👍 I have amended this to test expect(call.map(&:hearing_type)) to attempt to make this clearer